cgrok/grok_discover.c

```diff
@@ -186,6 +186,8 @@ void grok_discover(const grok_discover_t *gdt, /*grok_t *dest_grok, */
     if (max_matchlen == 0) { /* No valid matches were found */
       if (first_match_endpos > 0) {
         offset += first_match_endpos;
+      } else {
```

Comment:
We don't need `discover_main.c` for the library functionality, so I already pulled it out of this repo.